src/pdc-api.ts

```diff
@@ -13,28 +13,38 @@ const throwNotOk = (res: Response): Response => {
   return res;
 };
 
-const logError = (error: unknown, path: string) => {
-  logger.error({ error }, `Error fetching ${path}`);
+const logError = (error: unknown, path: string, params: URLSearchParams) => {
+  logger.error({ error, params }, `Error fetching ${path}`);
 };
 
 // Custom React hook to make authenticated requests to the configured API
-const usePdcApi = <T>(path: string): T | null => {
+const usePdcApi = <T>(
+  path: string,
+  params: URLSearchParams = new URLSearchParams(),
```

Comment:
Is there an advantage to `params` explicitly being a `URLSearchParams` here, instead of just an object? (As a TypeScript novice, I don't know if you're allowed to just say "it's an object, but we don't know what shape"; though of course we don't know the shape of the data passed to `URLSearchParams`, either.)

We'd have to add a line like `const searchParams = new URLSearchParams(params)` to the body of the hook, but the advantage would be a cleaner invocation of `usePdcApi()`.

Is that possible / a good idea?

Reply:
Definitely possible! In fact, I went back and forth on that point, and tried it both ways. The advantage of passing a `URLSearchParams` instead of a `Record<string, string>` is that the former allows for multiple instances of the same parameter, such as `?foo=bar&foo=baz`, which the latter cannot represent. In practice, that'd look like:

```ts
const params = new URLSearchParams();
params.append('foo', 'bar');
params.append('foo', 'baz');
const example = usePdcApi('/example', params);
```

That seems like it could be relevant for the someday future of faceted search, depending on how we end up encoding such queries into the API... but yes, I'll buy the argument that we don't need that complexity now, and could cross that bridge when we come to it.

Reply:
Oh, and actually now that I try a second time to write it this way, I remember the more compelling reason: using the object inside of the effect gets to be kind of a pain.

We can't use the object as one of the `useEffect` dependencies for the same reason we can't use the `URLSearchParams`: because [object literals are different on every render](https://react.dev/reference/react/useEffect#removing-unnecessary-object-dependencies). `.toString()` just returns `'[object Object]'`, so if we wanted to take the same string based approach we'd need to `JSON.stringify(params)`, which is somewhat feasible but feels far afield.

We could construct the `URLSearchParams` outside of the `useEffect` function, but then we need another name, and it feels slightly wrong to build the things needed for the effect outside of the effect. I say only "slightly" because we *are* doing that work outside of the effect in this approach, but it's far enough away that somehow it seems less bad, which is of course very silly.

> the advantage would be a cleaner invocation of `usePdcApi()`

Ultimately, where I landed is that `usePdcApi` is an internal helper, and the only callers of it should be in this file. (Side note: we shouldn't be exporting it, then!) In this case, a slightly more awkward invocation feels like a reasonable trade-off for a slightly more elegant body.

Reply:
> (Side note: we shouldn't be exporting [`usePdcApi`], then!)

Export removed in 0ff5d44!